gcp_delete_old_images_and_instances.py

```diff
@@ -0,0 +1,104 @@
+
+    # find the newest image for each family
+    newest_family_members = set()
+    for family in families:
+        newest_cmd = base_cmd + ['describe-from-family', '--format',
+                                 'object value(name)', family]
+        res = subprocess.run(newest_cmd, capture_output=True,
+                             check=True, text=True)
+        newest_family_members.add(res.stdout.strip())
+
+    # get all old images, including the newest image in a
+    # family (will be skipped below)
+    old_images_cmd = base_cmd + ['list', '--format', 'object value(name)',
+                                 '--no-standard-images', '--filter',
+                                 'creationTimestamp < -P2W']
+    res = subprocess.run(old_images_cmd, capture_output=True,
+                         check=True, text=True)
+    old_images = res.stdout.split()
+
+    # filter to-be-deleted images by the newest image in a family
+    delete_images = []
+    for old_image in old_images:
+        if old_image in newest_family_members:
+            print(f"not deleting {old_image}, it's the newest family member")
+        else:
+            delete_images.append(old_image)
+
+    if len(delete_images) == 0:
+        print("no images to delete")
+        sys.exit(0)
+
+    print("deleting images: ", ', '.join(delete_images))
+
+    # finally delete old images
+    delete_cmd = base_cmd + ['delete', '--quiet'] + delete_images
+    subprocess.run(delete_cmd, check=True)
+
+
+# Delete objects older than 23 hour
+def delete_objects(type: str):
+    base_cmd = ['gcloud', 'compute', '--project',
+                os.environ['GCP_PROJECT'], type]
+
+    # get objects older than 23 hours with their names and zones
+    # 23 hour is used instead of 1 day because cron job runs daily,
+    # so there could be race conditions
+    get_cmd = base_cmd + ['list', '--format', 'object value(name, zone)',
+                          '--filter', 'creationTimestamp < -P23H']
+    res = subprocess.run(get_cmd, capture_output=True,
+                         check=True, text=True)
+
+    delete_list = []
+    for line in res.stdout.strip().split('\n'):
+        # check if line is not empty
+        if line:
+            linesplit = line.split()
+            delete_list.append({'name': linesplit[0],
+                               'zone': linesplit[1]})
+
+    if delete_list:
+        print(f"{type} to be deleted: ", ', '.join(
+            disk['name'] for disk in delete_list))
+    else:
+        print(f"no {type} to delete")
+
+    for elem in delete_list:
+        print(f'Deleting {type} =', elem['name'])
+        delete_cmd = base_cmd + ['delete', elem['name'],
+                                 '--zone', elem['zone'], '--quiet']
+        subprocess.run(delete_cmd, check=True)
+
+
+def main():
+    if 'GCP_PROJECT' not in os.environ:
+        print("GCP_PROJECT not set", file=sys.stderr)
+        sys.exit(1)
+
+    delete_objects('instances')
+    delete_objects('disks')
```

Comment:
I think we should delete the instances / disks in a separate script. We only run the cleanup task if everything succeeded, but we are more likely to accumulate leftover things if there are failures. 

Also, if there are leftover objects, we might fail running due to resource constraints. So we should remove leftover things *before* we run packer.